docs/development/review.md

```diff
@@ -75,51 +133,69 @@ kindly,
+- Nr of Tests: phpunit: `./libs/composer/vendor/phpunit/phpunit/phpunit --list-tests <Path to Component Test Suite File>`
+- Nr of LoCs changed in last 365 days: `git diff --shortstat HEAD $(git log --since=365.days --oneline Services/WebDAV/ | tail -1 | cut -f 1 -d " ")^ <Path to Component>`
+- Nr of Commits in last 365 days: `git log --since=365.days --oneline <Path to Component> | wc -l`
+- To check the general code quality: `./CI/PHPStan/run_checks.sh <Path to Component`.
 
+#### If the file already exists:
+- Please update the sections 'Maintenance' and 'Statistics'.
+- Please add the sections 'Criteria based Review for [ilias_version]' und 'Qualitative Review for [ilias_version]'
+  **right below** the section 'Statistics'
 
+#### Review.md
 
+```
+# Review of `[name_of_component]` by [name_of_reviewer] at [date_of_review]
+
+## Maintenance
+- Maintainers/Coordinators:
+- Committers in last 365 days:
+
+## Statistics
+- Logical Lines of Code:
+- Nr of Tests:
+- LoCs changed in last 365 days:
+- Nr of Commits in last 365 days:
+- Nr of Open Mantis Issues:
+- Mantis Issues resolved in last 365 days:
+- Nr of Votes in last 365 days:
+- Nr of Votes for currently open issues:
+
+## Criteria based Review for [ilias_version]
+
+- [ ] `[name_of_component]`and its administration can be used in devmode with PHP 8 and 8.1 without producing PHP-specific errors
+- [ ] The code style is compliant with the [ILIAS Coding Style](https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/docs/development/coding-style.md)
+- Globals:
+	- [ ] The code contains no $_GET / $_POST / $_REQUEST / $_SESSION
+	- [ ] Access to $_GET-parameters happens through the request-wrapper and only in constructors
+	- [ ] Direct access to $_POST parameters through the request-wrapper is only used exceptionally and explained in a
+	      comment next to the access.
+- [ ] There are at least as many unit tests as in the last ILIAS Version and all tests pass. In cases where a feature is
+      completely removed from the ILIAS core or where code is centralized, the number of unit tests may exceptionally decrease.
+- [ ] All external libraries are up to date and compatible with PHP 8 and 8.1
+- Types:
+	- [ ] Explict PHP types are used (type hints, return types, typed properties) unless this is not possible due to
+	      missing support for Union-Types in PHP 8.
+	- [ ] The type `object` is not used
+	- [ ] `Mixed` is only used instead of a generic
+	- [ ] No named parameters are used
+- [ ] General Code Quality:
+    - [ ] There are no accesses to unknown classes
+    - [ ] There are no accesses to unknown methods
+    - [ ] There are no usages of undeclared class properties
+    - [ ] There is no dead/unreachable code
+    - [ ] There are no accesses to methods that are potentially missing (e.g. when Union Types or Magic Functions are used)
+    - [ ] All parameters and return values are either typed or their expected type is declared in a DocBlock, but not both
```

Comment:
I stumbled over this criterion. It says typing and DocBlock are not allowed at the same time. I often have an `array` typed as a parameter or return value and additionally make a more specified declaration in the DocBlock, like `@var int[]` or `@return int[]`. This should still be possible IMO.

Reply:
Hmm, valid point. Should we loosen this a bit? I think something similar would apply to `Iterator<$TYPE>`.

Reply:
Personally, I would stick to the allow-listing approach. I added the iterator to the list above. If we see it getting to long, I will start the part of my brain responsible for synthesis. Ok?